### PR TITLE
Prepare for 1.2.0 release

### PR DESCRIPTION
**Pull request by isc-bsaviano**

### CODE REVIEW

**Review by daimor — DISMISSED** [review]

File: CHANGELOG.md
Comment (gjsjohnmurray) [inline CHANGELOG.md:7]:
```suggestion
  - Preliminary web extension support (#782)
```

**Review by gjsjohnmurray — CHANGES_REQUESTED** [review]

**Review by gjsjohnmurray — APPROVED** [review]
LGTM, but maybe we need to wait until the issue reported today about not working with Insiders gets resolved. 1.63 started endgame yesterday and is likely to ship a week on Thursday.

**Comment by isc-bsaviano** [comment]
@gjsjohnmurray Yes, I think we need to wait until that issue is resolved

**Comment by gjsjohnmurray** [comment]
I propose we renumber this to 1.2. One reason is that #782 (github.dev and vscode.dev support) is a significant step. Another is I'd like us to be ready to use the upcoming extension pre-release mechanism (see https://github.com/microsoft/vscode/issues/138068). It's already working in 1.63 Insiders, which is now in endgame and expected to release next Thursday.

The recommended version numbering pattern for the pre-release mechanism is to use EVEN numbers in the minor position for the regular releases and ODD for pre-releases, so it'd be good to get the extension onto an even minor (2) soon.

Also pinging @isc-rsingh for an opinion on this.

**Comment by daimor** [comment]
I like the idea with pre-releases, so, it probably will be possible to produce our beta releases this way directly to marketplace. Beta-tests users will get new versions very quickly.

**Comment by isc-bsaviano** [comment]
@gjsjohnmurray Would it be possible to upload our beta versions in the pre-release stream using the current "-beta.X" suffix? If so, I think we should just modify that action to have a publish step and keep everything else as-is.

**Comment by daimor** [comment]
Marketplace at the moment allows only three parts, major.minor.patch
So, beta version number will go to patch part, and normal releases will get 0 there

**Comment by isidorn** [comment]
I see you are already looking into this, you can check out the docs which are work in progress and will be ready tomorrow
https://github.com/microsoft/vscode-docs/blob/vnext/api/working-with-extensions/publishing-extension.md#pre-release-extensions

**Comment by gjsjohnmurray** [comment]
> beta version number will go to patch part

An alternative strategy would be to use YYYYMMDDNN as the patch digit for pre-releases. This format means it's easier to differentiate pre-releases from regular ones, and to see at a glance how recent a particular pre-release is. The NN part gives scope for up to 99 pre-releases in a single day, which should be more than enough!

**Comment by isc-bsaviano** [comment]
@gjsjohnmurray @daimor Can you add the CI changes to this PR or should we open another one for that? Also, can pre-release versions published to the marketplace use the proposed API? If so, shouldn't we update our documentation?

**Comment by isc-bsaviano** [comment]
@daimor I will wait for the CI updates to be decided upon before I update and merge this PR because I will likely have to make README changes

**Comment by isc-rsingh** [comment]
I also agree with calling this 1.2 and all your reasoning behind it @gjsjohnmurray

**Pull request closed without merging**
